db/db_impl/db_impl.h

```diff
@@ -1240,6 +1240,38 @@ class DBImpl : public DB {
 
   std::atomic<bool> shutting_down_;
 
+  class RecoveryVersionEdits {
+   public:
+    ~RecoveryVersionEdits() {
+      for (size_t i = 0; i < edit_lists.size(); i++) {
+        for (size_t j = 0; j < edit_lists[i].size(); j++) {
+          delete edit_lists[i][j];
+        }
+        edit_lists[i].clear();
+      }
+      cfds.clear();
+      mutable_cf_opts.clear();
+      edit_lists.clear();
+    }
+
+    void UpdateVersionEdits(ColumnFamilyData* cfd, VersionEdit& _edit) {
+      if (map.find(cfd->GetID()) == map.end()) {
+        uint32_t size = static_cast<uint32_t>(map.size());
+        map.emplace(cfd->GetID(), size);
+        cfds.emplace_back(cfd);
+        mutable_cf_opts.emplace_back(cfd->GetLatestMutableCFOptions());
+        edit_lists.emplace_back(autovector<VersionEdit*>());
+      }
+      uint32_t i = map[cfd->GetID()];
+      edit_lists[i].emplace_back(new VersionEdit(_edit));
```

Comment:
When are the objects free'ed?

Reply:
In the destructor of this class https://github.com/facebook/rocksdb/pull/9634/files#diff-ce293088d5cc10fe3c3342c0968462651a9a955f6e7d745766b96132b1733801R1245